### ISSUE TITLE
Any reason why vercel_alias was removed after v10.0.1?

### ISSUE THREAD

**Issue by jacktan165** [chronark/terraform-provider-vercel#28]
I used vercel_alias to redirect www.website.com to website.com, after v10.0.1 that resource was removed. Is there an alternative to this?

**Comment by chronark** [comment]
Thank you.
I'm out for the weekend. Let me get back to this next week.

**Comment by chronark** [comment]
@jacktan165 I am so sorry, life was way too busy lately.

I reverted the alias resource because it got very confusing what vercel defines as domain and what they define as alias.
My first implementation was bound to cause a lot of confusion around that but I definitely see the need for a domain/alias resource.

1. There are account wide domains where you "register" a domain with vercel or set CNAMEs / name servers.
2. Then there are account wide aliases, which is the unique url part for each deployment
3. And finally you can add a domain to each project down the the individual branch 

I believe you want the 3rd option, correct?

I think ideally we'd implement a repeatable field on the project resource to set domains, what do you think about something like this?

```terraform
resource "vercel_project" "my_project" {
  name = "..."
  ...
  domain {
    // Target destination domain for redirect.
    redirect = "example.com" 

    // optional: defaults to 200
    redirectStatusCode = 200 // | The redirect status code (301, 302, 307, 308).

    // optional:
    gitBranch: "main"
  }
}

```

**Comment by recurser** [comment]
@chronark +1 to this - I've dropped back to `v0.10.0` in the meantime. What you have suggested above looks pretty good to me 👍

**Comment by chronark** [comment]
@recurser @jacktan165
I have released v0.10.5-rc.1 which adds the `domain` field on projects
https://registry.terraform.io/providers/chronark/vercel/latest

```terraform
resource "vercel_project" "my_project" {
  name = "..."
  ...
  domain {
    name = "chronark.com"
    }
}

```

I was running into some issues when adding redirect and redirectStatusCode and it's still lacking tests, but I thought I'd rather push this out and come back to them as soon as I can.

Let me know how that works for you.

**Comment by recurser** [comment]
Thanks for the quick work @chronark ! `v0.10.5-rc.1` *appeared* to apply ok, but when I checked Vercel the domain wasn't actually added:

<img width="805" alt="Screen Shot 2021-08-12 at 21 22 36" src="https://user-images.githubusercontent.com/96322/129195933-d3edc3ab-3fb1-424c-b7e9-eb8498b423fa.png">

Any ideas, or something I can try to help debug it?

**Comment by chronark** [comment]
Just so we are on the same page, can you run the following terraform configuration please:
```terraform
terraform {
  required_providers {
    vercel = {
      source = "chronark/vercel"
      version = "0.10.5-rc.1"
    }
  }
}

provider "vercel" {
}



resource "vercel_project" "my_project" {
  name = "test"
  git_repository {
    type = "github"
    repo = "REPLACE_ME"
  }

  domain {
    name = "REPLACE_ME"
  }


}

```
```
terraform init
VERCEL_TOKEN=XXX terraform apply
```
That does work for me.

![image](https://user-images.githubusercontent.com/18246773/129202307-464a3ab6-aa98-44de-8452-74f47e0bdfb6.png)


Also could you post the `terraform.tfstate` file afterwards, just to make sure terraform knows about the domain as well.
The domain should be under `resources[x].instances[y].attributes.domain`

**Comment by chronark** [comment]
Also just to rule out any confusion: This can only add a domain to a project when the domain has been added to your vercel account/team previously.

Adding the domain to your account is a different thing and can be done with the `vercel_domain` resource.

**Comment by recurser** [comment]
@chronark thanks for getting back so quickly. That does work with a brand new standalone terraform configuration as you described, adding the domain I want. It doesn't add anything in my actual development terraform configuration though, which is more complex. I've got something like this:

```
resource "vercel_project" "dashboard" {
  depends_on    = [vercel_domain.main, cloudflare_record.dashboard]
  framework     = "nextjs"
  name          = "${local.prefix}-dashboard-${var.environment}"
  public_source = false

  domain {
    git_branch = "develop"
    name       = "dashboard-test.dave.host"
  }

  git_repository {
    type = "github"
    repo = "recurser/xyz"
  }
}

resource "vercel_domain" "main" {
  name = "dave.host"
}
```

The `vercel_domain` part works fine, and the domain has been added to the account (and the standalone terraform configuration working confirms this). I'll do some digging and see if I can figure it out...

**Comment by chronark** [comment]
Can you try it without the git_branch?
That didn't work properly when I hacked it together.

**Comment by recurser** [comment]
@chronark after a bit of digging - this works when you create a **new** project, but doesn't work with an existing project (as is the case for me) where I have added the `domain { }` config later. Removing the `git_branch` doesn't help.

Looking at the provider code, I guess it's because it's implemented in `resourceProjectCreate` but not `resourceProjectUpdate`?

**Comment by chronark** [comment]
Yeah I think you are right, It also needs to be implemented on the update method.
EIther that or force recreating the entire project which is not really desired.

Do you think you could implement it?
I think you only need to copy over the code from the `resourceProjectCreate` method

**Comment by recurser** [comment]
@chronark sounds good, I'll give it a go 👍

**Comment by chronark** [comment]
Awesome, let me know if you run into issues and I'll help you out.

**Comment by recurser** [comment]
@chronark thanks - the first issue I've run into that I can't figure out is how to log debugging info to Terraform output. I've tried adding `log.Println("[ERROR] Something happened!")` as [suggested by the Terraform docs](https://www.terraform.io/docs/extend/debugging.html), but I can't see the output anywhere. Do you have any hints on that? I've never actually worked on a Terraform provider before so I'm up against a bit of a learning curve 🙇‍♂️

**Comment by chronark** [comment]
@recurser Haha yeah I still haven't figured out how to actually log something. I have resorted to `panic`ing. At least that worked and I could print something, but it's far from ideal.

**Comment by recurser** [comment]
@chronark that's a big help, thanks (and `panic` is a good idea) 👍 I'll have a go and get back to you if I get stuck.
